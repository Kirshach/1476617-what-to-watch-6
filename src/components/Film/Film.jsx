import React, {useEffect} from 'react';
import {connect} from 'react-redux';
import {Link, useParams} from 'react-router-dom';
import PropTypes from 'prop-types';

import Description from './Description';
import Details from './Details';
import MovieCardList from '../MovieCardList/MovieCardList';
import ReviewsList from './ReviewsList';
import TabBar from './TabBar';

import {filmArrayPropTypes} from '../../prop-types/film';
import {fetchFilmThunk} from '../../store/domain/thunks';

import {useNavigation} from '../../hooks';

import LoadingPlaceholder from '../LoadingPlaceholder/LoadingPlaceholder';
import PageFooter from '../PageFooter/PageFooter';
import PageHeader from '../PageHeader/PageHeader';

import {FilmAppSubroutes} from '../../const';

const TABS = [
  {label: `Overview`, route: FilmAppSubroutes.overview},
  {label: `Details`, route: FilmAppSubroutes.details},
  {label: `Reviews`, route: FilmAppSubroutes.reviews},
];

const getFilmPageBody = (film, tab) => {
  switch (tab) {
    case FilmAppSubroutes.overview:
      return <Description film={film}/>;
    case FilmAppSubroutes.details:
      return <Details film={film}/>;
    case FilmAppSubroutes.reviews:
      return <ReviewsList id={film.id}/>;
    default: throw new Error(`Invalid route provided to getFilmPageBody function`);
  }
};

const getSimilarFilms = (films, film) => {
  return films && films.filter(
      (anotherFilm) => anotherFilm.genre === film.genre && anotherFilm.id !== film.id
  ) || [];
};

const Film = ({
  dispatchFetchFilmThunk,
  film,
  films,
  filmHasLoaded,
  filmsHaveLoaded,
  isAuthorized,
}) => {
  const {redirect} = useNavigation();
  const {id, tab = ``} = useParams();
  const similarFilms = getSimilarFilms(films, film);

  useEffect(() => {
    dispatchFetchFilmThunk(id);
  }, [id]);

  return filmHasLoaded
    ? (
      <>
        <section className="movie-card movie-card--full">
          <div className="movie-card__hero">
            <div className="movie-card__bg">
              <img src={film.backgroundImage} alt={film.name} />
            </div>

            <h1 className="visually-hidden">WTW</h1>

            <PageHeader className="movie-card__head"/>

            <div className="movie-card__wrap">
              <div className="movie-card__desc">
                <h2 className="movie-card__title">{film.name}</h2>
                <p className="movie-card__meta">
                  <span className="movie-card__genre">{film.genre}</span>
                  <span className="movie-card__year">{film.released}</span>
                </p>

                <div className="movie-card__buttons">
                  <button className="btn btn--play movie-card__button" type="button" onClick={() => redirect(`/player/${film.id}`)}>
                    <svg viewBox="0 0 19 19" width="19" height="19">
                      <use xlinkHref="#play-s"></use>
                    </svg>
                    <span>Play</span>
                  </button>
                  <button className="btn btn--list movie-card__button" type="button">
                    <svg viewBox="0 0 19 20" width="19" height="20">
                      <use xlinkHref="#add"></use>
                    </svg>
                    <span>My list</span>
                  </button>

                  {isAuthorized &&
                    <Link to={`/films/${film.id}/review`} className="btn movie-card__button">
                      Add review
                    </Link>}

                </div>
              </div>
            </div>
          </div>

          <div className="movie-card__wrap movie-card__translate-top">
            <div className="movie-card__info">
              <div className="movie-card__poster movie-card__poster--big">
                <img src={film.posterImage} alt={`${film.name} poster`} width="218" height="327" />
              </div>

              <div className="movie-card__desc">

                <TabBar tabs={TABS}/>

                {getFilmPageBody(film, tab)}
              </div>
            </div>
          </div>
        </section>

        <div className="page-content">

          <section className="catalog catalog--like-this">
            {similarFilms.length > 0 &&
              <>
                <h2 className="catalog__title">More like this</h2>
                <div className="catalog__movies-list">
                  {filmsHaveLoaded
                    ? <MovieCardList films={similarFilms} />
                    : <LoadingPlaceholder />}
                </div>
              </>}
          </section>

          <PageFooter />
        </div>
      </>
    ) : (
      <LoadingPlaceholder />
    );
};

Film.propTypes = {
  dispatchFetchFilmThunk: PropTypes.func.isRequired,
  film: PropTypes.object.isRequired,
  films: filmArrayPropTypes,
  filmHasLoaded: PropTypes.bool.isRequired,
  filmsHaveLoaded: PropTypes.bool.isRequired,
  isAuthorized: PropTypes.bool.isRequired,
};

const mapStateToProps = (state) => ({
  film: state.domain.film,
  films: state.domain.films,
  filmHasLoaded: state.app.filmHasLoaded,
  filmsHaveLoaded: state.app.filmsHaveLoaded,
  isAuthorized: state.app.isAuthorized,
});

const mapDispatchToProps = (dispatch) => ({
  dispatchFetchFilmThunk: (id) => dispatch(fetchFilmThunk(id)),
});


const FilmWithStore = connect(
    mapStateToProps,
    mapDispatchToProps
)(Film);

export default FilmWithStore;
