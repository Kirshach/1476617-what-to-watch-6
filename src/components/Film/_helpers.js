import React from 'react';

import Description from './Description';
import ReviewsList from './ReviewsList';
import Details from './Details';

import {AppRoutes} from '../../const';
import {Subroutes} from './_const';

export const getFilmPageBody = (film, tab) => {
  switch (tab) {
    case Subroutes.overview:
      return <Description film={film}/>;
    case Subroutes.details:
      return <Details film={film}/>;
    case Subroutes.reviews:
      return <ReviewsList id={film.id}/>;
    default: throw new Error(`Invalid route provided to getFilmPageBody function`);
  }
};
export const getSimilarFilms = (films, film) => {
  return films && films.filter(
      (anotherFilm) => anotherFilm.genre === film.genre && anotherFilm.id !== film.id
  ).slice(0, 4) || [];
};
export const formatDuration = (time) => {
  const hours = Math.floor(time / 60);
  const minutes = time % 60;
  return hours > 0 ? `${hours}h ${minutes}m` : `${minutes}m`;
};
export const getPlayerRoute = (id) => `${AppRoutes.MOVIE_PLAYER_BASE_ROUTE}/${id}`;
export const getTabClassName = (isCurrentRoute) => `movie-nav__item ${isCurrentRoute ? `movie-nav__item--active` : ``}`;
export const getVerboseRating = (score) => {
  let verboseRating;
  if (score < 0 || score > 10) {
    throw new Error(`Unknown rating ${score} received`);
  } else if (score < 3) {
    verboseRating = `Bad`;
  } else if (score < 5) {
    verboseRating = `Normal`;
  } else if (score < 8) {
    verboseRating = `Good`;
  } else if (score < 10) {
    verboseRating = `Very Good`;
  } else if (score === 10) {
    verboseRating = `Awesome`;
  }
  return verboseRating;
};
export const formatRating = (rating) => rating.toFixed(1);
