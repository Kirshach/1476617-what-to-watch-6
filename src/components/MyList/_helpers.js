import React from 'react';

import MovieCardList from '../MovieCardList/MovieCardList';
import LoadingPlaceholder from '../LoadingPlaceholder/LoadingPlaceholder';

export const getFavouriteMoviesListComponent = (
    favouriteFilms,
    favouriteFilmsHaveLoaded,
) => {
  if (favouriteFilmsHaveLoaded) {
    return favouriteFilms.length > 0
      ? <MovieCardList films={favouriteFilms} />
      : (
        <div className="no-films-placeholder">
          There seems to be no sign of life in your favourite movies list
        </div>
      );
  } else {
    return <LoadingPlaceholder />;
  }
};
