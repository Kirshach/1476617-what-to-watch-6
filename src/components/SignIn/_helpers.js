import {FIELD_NAMES} from './_const';

const emailRegExp = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;

export const validate = (name, value) => {
  switch (name) {
    case FIELD_NAMES.EMAIL:
      if (!emailRegExp.test(value)) {
        return `The e-mail is invalid`;
      } else {
        return undefined;
      }
    case FIELD_NAMES.PASSWORD:
      if (value.length < 1) {
        return `Well, at least enter A SINGLE CHARACTER in the password field :P`;
      } else {
        return undefined;
      }
    default:
      return undefined;
  }
};
