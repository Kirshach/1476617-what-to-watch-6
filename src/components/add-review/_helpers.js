import {FIELD_NAMES} from './_const';

export const validate = (name, value) => {
  switch (name) {
    case FIELD_NAMES.COMMENT:
      if (value.length < 50) {
        return `Comment should be at least 50 characters long.`;
      } else if (value.length > 400) {
        return `Comment should be at most 400 characters long.`;
      } else {
        return undefined;
      }
    case FIELD_NAMES.RATING: {
      if (value === null) {
        return `Film should be rated`;
      } else {
        const numValue = Number(value);
        if (!(Number.isInteger(numValue) || numValue > 10 || numValue < 0)) {
          return `Unknown rating value`;
        } else {
          return undefined;
        }
      }
    }
    default:
      return undefined;
  }
};

export const formatValues = (values) => {
  return {
    ...values,
    [FIELD_NAMES.RATING]: Number(values[FIELD_NAMES.RATING])
  };
};
