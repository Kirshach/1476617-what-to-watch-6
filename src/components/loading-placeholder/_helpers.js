export const getNewDots = (dots) => dots.length >= 3 ? `.` : dots + `.`;
