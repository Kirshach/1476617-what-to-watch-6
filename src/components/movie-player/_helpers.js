const dd = (num) => {
  return num > 9 ? num : `0` + num;
};

export const formatDuration = (timeInSeconds, withHours = false) => {
  if (timeInSeconds === null || timeInSeconds === undefined) {
    return null;
  }
  const h = Math.floor(timeInSeconds / 3600);
  const m = Math.floor(timeInSeconds / 60);
  const s = Math.round(timeInSeconds % 60);

  return (h > 0 || withHours)
    ? [h, dd(m), dd(s)].join(`:`)
    : [m, dd(s)].join(`:`);
};

export const getTogglerRelativePos = (event, progressBarRef) => {
  const progressBarBounds = progressBarRef.current.getBoundingClientRect();
  const progressBarWidth = progressBarBounds.right - progressBarBounds.left;
  const posX = event.clientX - progressBarBounds.left;
  let posXrelative = posX / progressBarWidth * 100;

  return Math.max(Math.min(posXrelative, 100), 0);
};

export const getTimestamp = (duration, posXrelative) => duration * posXrelative / 100;
