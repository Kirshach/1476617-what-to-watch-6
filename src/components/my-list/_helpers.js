import React from 'react';

import MovieCardList from '../movie-card-list/movie-card-list';
import LoadingPlaceholder from '../loading-placeholder/loading-placeholder';

export const getFavouriteMoviesListComponent = (
    favouriteFilms,
    favouriteFilmsHaveLoaded,
) => {
  if (favouriteFilmsHaveLoaded) {
    return favouriteFilms.length > 0
      ? <MovieCardList films={favouriteFilms} />
      : (
        <div className="no-films-placeholder">
          There seems to be no sign of life in your favourite movies list
        </div>
      );
  } else {
    return <LoadingPlaceholder />;
  }
};
