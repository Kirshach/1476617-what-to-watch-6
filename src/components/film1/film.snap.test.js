import React from 'react';
import {useDispatch, useSelector} from 'react-redux';
import {MemoryRouter, Route} from 'react-router';
import {render} from '@testing-library/react';

import {store} from '../../mocks/store/store';
import {useFilm} from '../../hooks/use-film';
import {macbeth} from '../../mocks/film';
import {films} from '../../mocks/films';
import {AppRoutes} from '../../const';

import * as helpers from './_helpers';

import Film from './film';

function MockVideoPlayer() {
  return <div>This is a VideoPlayer placeholder</div>;
}

jest.mock(`react-redux`);
jest.mock(`../../hooks/use-film`);
jest.spyOn(helpers, `getSimilarFilms`).mockReturnValue(films);
jest.mock(`../video-player/video-player`, () => ({
  __esModule: true,
  default: MockVideoPlayer,
}));

describe(`<Film />`, () => {
  useDispatch.mockReturnValue(() => Promise.resolve());
  useSelector.mockImplementation((selector) => selector(store));
  useFilm.mockReturnValue([macbeth]);

  test(`renders main overview tab correctly`, () => {
    const {id} = macbeth;
    const path = AppRoutes.getFilmRoute(id);

    const {container} = render(
        <MemoryRouter initialEntries={[path]}>
          <Route path={AppRoutes.FILM}>
            <Film film={macbeth} />
          </Route>
        </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });

  test(`renders details tab correctly`, () => {
    const {id} = macbeth;
    const path = AppRoutes.getFilmDetailsRoute(id);

    const {container} = render(
        <MemoryRouter initialEntries={[path]}>
          <Route path={AppRoutes.FILM}>
            <Film film={macbeth} />
          </Route>
        </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });

  test(`renders reviews tab correctly`, () => {
    const {id} = macbeth;
    const path = AppRoutes.getFilmReviewsRoute(id);

    const {container} = render(
        <MemoryRouter initialEntries={[path]}>
          <Route path={AppRoutes.FILM}>
            <Film film={macbeth} />
          </Route>
        </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });

});
