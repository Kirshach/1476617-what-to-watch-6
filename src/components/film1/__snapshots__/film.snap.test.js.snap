// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Film /> renders details tab correctly 1`] = `
<div>
  <section
    class="movie-card movie-card--full"
  >
    <div
      class="movie-card__hero"
    >
      <div
        class="movie-card__bg"
      >
        <img
          alt="Macbeth"
          src="https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Macbeth.jpg"
        />
      </div>
      <h1
        class="visually-hidden"
      >
        WTW
      </h1>
      <header
        class="page-header movie-card__head"
      >
        <div
          class="logo"
        >
          <a
            class="logo__link"
            href="/"
          >
            <span
              class="logo__letter logo__letter--1"
            >
              W
            </span>
            <span
              class="logo__letter logo__letter--2"
            >
              T
            </span>
            <span
              class="logo__letter logo__letter--3"
            >
              W
            </span>
          </a>
        </div>
        <div
          class="user-block"
        >
          <a
            class="user-block__link"
            href="/mylist"
          >
            <div
              class="user-block__avatar"
            >
              <img
                alt="User avatar"
                height="63"
                src="https://assets.htmlacademy.ru/intensives/javascript-3/avatar/6.jpg"
                width="63"
              />
            </div>
          </a>
        </div>
      </header>
      <div
        class="movie-card__wrap"
      >
        <div
          class="movie-card__desc"
        >
          <h2
            class="movie-card__title"
          >
            Macbeth
          </h2>
          <p
            class="movie-card__meta"
          >
            <span
              class="movie-card__genre"
            >
              Drama
            </span>
            <span
              class="movie-card__year"
            >
              2015
            </span>
          </p>
          <div
            class="movie-card__buttons"
          >
            <button
              class="btn btn--play movie-card__button"
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlink:href="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </button>
            <button
              class="btn btn--list movie-card__button"
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlink:href="#in-list"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
            <a
              class="btn movie-card__button"
              href="/films/1/review"
            >
              Add review
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      class="movie-card__wrap movie-card__translate-top"
    >
      <div
        class="movie-card__info"
      >
        <div
          class="movie-card__poster movie-card__poster--big"
        >
          <img
            alt="Macbeth poster"
            height="327"
            src="https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Macbeth.jpg"
            width="218"
          />
        </div>
        <div
          class="movie-card__desc"
        >
          <nav
            class="movie-nav movie-card__nav"
          >
            <ul
              class="movie-nav__list"
            >
              <li
                class="movie-nav__item "
              >
                <a
                  class="movie-nav__link"
                  href="/films/1/"
                >
                  Overview
                </a>
              </li>
              <li
                class="movie-nav__item movie-nav__item--active"
              >
                <a
                  class="movie-nav__link"
                  href="/films/1/details"
                >
                  Details
                </a>
              </li>
              <li
                class="movie-nav__item "
              >
                <a
                  class="movie-nav__link"
                  href="/films/1/reviews"
                >
                  Reviews
                </a>
              </li>
            </ul>
          </nav>
          <div
            class="movie-card__text movie-card__row"
          >
            <div
              class="movie-card__text-col"
            >
              <p
                class="movie-card__details-item"
              >
                <strong
                  class="movie-card__details-name"
                >
                  Director
                </strong>
                <span
                  class="movie-card__details-value"
                >
                  Justin Kurzel
                </span>
              </p>
              <p
                class="movie-card__details-item"
              >
                <strong
                  class="movie-card__details-name"
                >
                  Starring
                </strong>
                <span
                  class="movie-card__details-value"
                >
                  Michael Fassbender
                  , 
                  <br />
                  Marion Cotillard
                  , 
                  <br />
                  Jack Madigan
                </span>
              </p>
            </div>
            <div
              class="movie-card__text-col"
            >
              <p
                class="movie-card__details-item"
              >
                <strong
                  class="movie-card__details-name"
                >
                  Run Time
                </strong>
                <span
                  class="movie-card__details-value"
                >
                  1h 53m
                </span>
              </p>
              <p
                class="movie-card__details-item"
              >
                <strong
                  class="movie-card__details-name"
                >
                  Genre
                </strong>
                <span
                  class="movie-card__details-value"
                >
                  Drama
                </span>
              </p>
              <p
                class="movie-card__details-item"
              >
                <strong
                  class="movie-card__details-name"
                >
                  Released
                </strong>
                <span
                  class="movie-card__details-value"
                >
                  2015
                </span>
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
  <div
    class="page-content"
  >
    <section
      class="catalog catalog--like-this"
    >
      <h2
        class="catalog__title"
      >
        More like this
      </h2>
      <div
        class="catalog__movies-list"
      >
        <article
          class="small-movie-card catalog__movies-card"
          style="cursor: pointer;"
        >
          <div
            class="small-movie-card__image"
          >
            <div>
              This is a VideoPlayer placeholder
            </div>
          </div>
          <h3
            class="small-movie-card__title"
            style="pointer-events: none;"
          >
            <span
              class="small-movie-card__link"
            >
              Bronson
            </span>
          </h3>
        </article>
        <article
          class="small-movie-card catalog__movies-card"
          style="cursor: pointer;"
        >
          <div
            class="small-movie-card__image"
          >
            <div>
              This is a VideoPlayer placeholder
            </div>
          </div>
          <h3
            class="small-movie-card__title"
            style="pointer-events: none;"
          >
            <span
              class="small-movie-card__link"
            >
              The Revenant
            </span>
          </h3>
        </article>
        <article
          class="small-movie-card catalog__movies-card"
          style="cursor: pointer;"
        >
          <div
            class="small-movie-card__image"
          >
            <div>
              This is a VideoPlayer placeholder
            </div>
          </div>
          <h3
            class="small-movie-card__title"
            style="pointer-events: none;"
          >
            <span
              class="small-movie-card__link"
            >
              Macbeth
            </span>
          </h3>
        </article>
      </div>
    </section>
    <footer
      class="page-footer"
    >
      <div
        class="logo"
      >
        <a
          class="logo__link"
          href="/"
        >
          <span
            class="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            class="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            class="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        class="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>
</div>
`;

exports[`<Film /> renders main overview tab correctly 1`] = `
<div>
  <section
    class="movie-card movie-card--full"
  >
    <div
      class="movie-card__hero"
    >
      <div
        class="movie-card__bg"
      >
        <img
          alt="Macbeth"
          src="https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Macbeth.jpg"
        />
      </div>
      <h1
        class="visually-hidden"
      >
        WTW
      </h1>
      <header
        class="page-header movie-card__head"
      >
        <div
          class="logo"
        >
          <a
            class="logo__link"
            href="/"
          >
            <span
              class="logo__letter logo__letter--1"
            >
              W
            </span>
            <span
              class="logo__letter logo__letter--2"
            >
              T
            </span>
            <span
              class="logo__letter logo__letter--3"
            >
              W
            </span>
          </a>
        </div>
        <div
          class="user-block"
        >
          <a
            class="user-block__link"
            href="/mylist"
          >
            <div
              class="user-block__avatar"
            >
              <img
                alt="User avatar"
                height="63"
                src="https://assets.htmlacademy.ru/intensives/javascript-3/avatar/6.jpg"
                width="63"
              />
            </div>
          </a>
        </div>
      </header>
      <div
        class="movie-card__wrap"
      >
        <div
          class="movie-card__desc"
        >
          <h2
            class="movie-card__title"
          >
            Macbeth
          </h2>
          <p
            class="movie-card__meta"
          >
            <span
              class="movie-card__genre"
            >
              Drama
            </span>
            <span
              class="movie-card__year"
            >
              2015
            </span>
          </p>
          <div
            class="movie-card__buttons"
          >
            <button
              class="btn btn--play movie-card__button"
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlink:href="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </button>
            <button
              class="btn btn--list movie-card__button"
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlink:href="#in-list"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
            <a
              class="btn movie-card__button"
              href="/films/1/review"
            >
              Add review
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      class="movie-card__wrap movie-card__translate-top"
    >
      <div
        class="movie-card__info"
      >
        <div
          class="movie-card__poster movie-card__poster--big"
        >
          <img
            alt="Macbeth poster"
            height="327"
            src="https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Macbeth.jpg"
            width="218"
          />
        </div>
        <div
          class="movie-card__desc"
        >
          <nav
            class="movie-nav movie-card__nav"
          >
            <ul
              class="movie-nav__list"
            >
              <li
                class="movie-nav__item movie-nav__item--active"
              >
                <a
                  class="movie-nav__link"
                  href="/films/1/"
                >
                  Overview
                </a>
              </li>
              <li
                class="movie-nav__item "
              >
                <a
                  class="movie-nav__link"
                  href="/films/1/details"
                >
                  Details
                </a>
              </li>
              <li
                class="movie-nav__item "
              >
                <a
                  class="movie-nav__link"
                  href="/films/1/reviews"
                >
                  Reviews
                </a>
              </li>
            </ul>
          </nav>
          <div
            class="movie-rating"
          >
            <div
              class="movie-rating__score"
            >
              3.3
            </div>
            <p
              class="movie-rating__meta"
            >
              <span
                class="movie-rating__level"
              >
                Normal
              </span>
              <span
                class="movie-rating__count"
              >
                48798
                 ratings
              </span>
            </p>
          </div>
          <div
            class="movie-card__text"
          >
            <p>
              Macbeth, the Thane of Glamis, receives a prophecy from a trio of witches that one day he will become King of Scotland. Consumed by ambition and spurred to action by his wife, Macbeth murders his king and takes the throne for himself.
            </p>
            <p
              class="movie-card__director"
            >
              <strong>
                Director: 
                Justin Kurzel
              </strong>
            </p>
            <p
              class="movie-card__starring"
            >
              <strong>
                Starring: 
                Michael Fassbender, Marion Cotillard, Jack Madigan
                 and others
              </strong>
            </p>
          </div>
        </div>
      </div>
    </div>
  </section>
  <div
    class="page-content"
  >
    <section
      class="catalog catalog--like-this"
    >
      <h2
        class="catalog__title"
      >
        More like this
      </h2>
      <div
        class="catalog__movies-list"
      >
        <article
          class="small-movie-card catalog__movies-card"
          style="cursor: pointer;"
        >
          <div
            class="small-movie-card__image"
          >
            <div>
              This is a VideoPlayer placeholder
            </div>
          </div>
          <h3
            class="small-movie-card__title"
            style="pointer-events: none;"
          >
            <span
              class="small-movie-card__link"
            >
              Bronson
            </span>
          </h3>
        </article>
        <article
          class="small-movie-card catalog__movies-card"
          style="cursor: pointer;"
        >
          <div
            class="small-movie-card__image"
          >
            <div>
              This is a VideoPlayer placeholder
            </div>
          </div>
          <h3
            class="small-movie-card__title"
            style="pointer-events: none;"
          >
            <span
              class="small-movie-card__link"
            >
              The Revenant
            </span>
          </h3>
        </article>
        <article
          class="small-movie-card catalog__movies-card"
          style="cursor: pointer;"
        >
          <div
            class="small-movie-card__image"
          >
            <div>
              This is a VideoPlayer placeholder
            </div>
          </div>
          <h3
            class="small-movie-card__title"
            style="pointer-events: none;"
          >
            <span
              class="small-movie-card__link"
            >
              Macbeth
            </span>
          </h3>
        </article>
      </div>
    </section>
    <footer
      class="page-footer"
    >
      <div
        class="logo"
      >
        <a
          class="logo__link"
          href="/"
        >
          <span
            class="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            class="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            class="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        class="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>
</div>
`;

exports[`<Film /> renders reviews tab correctly 1`] = `
<div>
  <section
    class="movie-card movie-card--full"
  >
    <div
      class="movie-card__hero"
    >
      <div
        class="movie-card__bg"
      >
        <img
          alt="Macbeth"
          src="https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Macbeth.jpg"
        />
      </div>
      <h1
        class="visually-hidden"
      >
        WTW
      </h1>
      <header
        class="page-header movie-card__head"
      >
        <div
          class="logo"
        >
          <a
            class="logo__link"
            href="/"
          >
            <span
              class="logo__letter logo__letter--1"
            >
              W
            </span>
            <span
              class="logo__letter logo__letter--2"
            >
              T
            </span>
            <span
              class="logo__letter logo__letter--3"
            >
              W
            </span>
          </a>
        </div>
        <div
          class="user-block"
        >
          <a
            class="user-block__link"
            href="/mylist"
          >
            <div
              class="user-block__avatar"
            >
              <img
                alt="User avatar"
                height="63"
                src="https://assets.htmlacademy.ru/intensives/javascript-3/avatar/6.jpg"
                width="63"
              />
            </div>
          </a>
        </div>
      </header>
      <div
        class="movie-card__wrap"
      >
        <div
          class="movie-card__desc"
        >
          <h2
            class="movie-card__title"
          >
            Macbeth
          </h2>
          <p
            class="movie-card__meta"
          >
            <span
              class="movie-card__genre"
            >
              Drama
            </span>
            <span
              class="movie-card__year"
            >
              2015
            </span>
          </p>
          <div
            class="movie-card__buttons"
          >
            <button
              class="btn btn--play movie-card__button"
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlink:href="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </button>
            <button
              class="btn btn--list movie-card__button"
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlink:href="#in-list"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
            <a
              class="btn movie-card__button"
              href="/films/1/review"
            >
              Add review
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      class="movie-card__wrap movie-card__translate-top"
    >
      <div
        class="movie-card__info"
      >
        <div
          class="movie-card__poster movie-card__poster--big"
        >
          <img
            alt="Macbeth poster"
            height="327"
            src="https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Macbeth.jpg"
            width="218"
          />
        </div>
        <div
          class="movie-card__desc"
        >
          <nav
            class="movie-nav movie-card__nav"
          >
            <ul
              class="movie-nav__list"
            >
              <li
                class="movie-nav__item "
              >
                <a
                  class="movie-nav__link"
                  href="/films/1/"
                >
                  Overview
                </a>
              </li>
              <li
                class="movie-nav__item "
              >
                <a
                  class="movie-nav__link"
                  href="/films/1/details"
                >
                  Details
                </a>
              </li>
              <li
                class="movie-nav__item movie-nav__item--active"
              >
                <a
                  class="movie-nav__link"
                  href="/films/1/reviews"
                >
                  Reviews
                </a>
              </li>
            </ul>
          </nav>
          <div
            class="movie-card__reviews movie-card__row"
          >
            <div
              class="movie-card__reviews-col"
            >
              <div
                class="review"
              >
                <blockquote
                  class="review__quote"
                >
                  <p
                    class="review__text"
                  >
                    Poised to be an instant classic, almost everything about this film is phenomenal - the acting, the cinematography, the discography, etc.
                  </p>
                  <footer
                    class="review__details"
                  >
                    <cite
                      class="review__author"
                    >
                      Corey
                    </cite>
                    <time
                      class="review__date"
                      datetime="2021-02-15T08:04:28.658Z"
                    >
                      February 15, 2021
                    </time>
                  </footer>
                </blockquote>
                <div
                  class="review__rating"
                >
                  7,8
                </div>
              </div>
              <div
                class="review"
              >
                <blockquote
                  class="review__quote"
                >
                  <p
                    class="review__text"
                  >
                    A movie that will take you to another world full of emotions.
                  </p>
                  <footer
                    class="review__details"
                  >
                    <cite
                      class="review__author"
                    >
                      Jack
                    </cite>
                    <time
                      class="review__date"
                      datetime="2021-02-11T08:04:28.658Z"
                    >
                      February 11, 2021
                    </time>
                  </footer>
                </blockquote>
                <div
                  class="review__rating"
                >
                  5,9
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
  <div
    class="page-content"
  >
    <section
      class="catalog catalog--like-this"
    >
      <h2
        class="catalog__title"
      >
        More like this
      </h2>
      <div
        class="catalog__movies-list"
      >
        <article
          class="small-movie-card catalog__movies-card"
          style="cursor: pointer;"
        >
          <div
            class="small-movie-card__image"
          >
            <div>
              This is a VideoPlayer placeholder
            </div>
          </div>
          <h3
            class="small-movie-card__title"
            style="pointer-events: none;"
          >
            <span
              class="small-movie-card__link"
            >
              Bronson
            </span>
          </h3>
        </article>
        <article
          class="small-movie-card catalog__movies-card"
          style="cursor: pointer;"
        >
          <div
            class="small-movie-card__image"
          >
            <div>
              This is a VideoPlayer placeholder
            </div>
          </div>
          <h3
            class="small-movie-card__title"
            style="pointer-events: none;"
          >
            <span
              class="small-movie-card__link"
            >
              The Revenant
            </span>
          </h3>
        </article>
        <article
          class="small-movie-card catalog__movies-card"
          style="cursor: pointer;"
        >
          <div
            class="small-movie-card__image"
          >
            <div>
              This is a VideoPlayer placeholder
            </div>
          </div>
          <h3
            class="small-movie-card__title"
            style="pointer-events: none;"
          >
            <span
              class="small-movie-card__link"
            >
              Macbeth
            </span>
          </h3>
        </article>
      </div>
    </section>
    <footer
      class="page-footer"
    >
      <div
        class="logo"
      >
        <a
          class="logo__link"
          href="/"
        >
          <span
            class="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            class="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            class="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        class="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>
</div>
`;
